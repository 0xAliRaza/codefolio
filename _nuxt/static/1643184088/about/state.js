window.__NUXT__=(function(a){return {staticAssetsBase:"\u002Fcodefolio\u002F_nuxt\u002Fstatic\u002F1643184088",layout:"default",error:a,serverRendered:true,routePath:"\u002Fabout",config:{_app:{basePath:"\u002Fcodefolio\u002F",assetsPath:"\u002Fcodefolio\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"7ccc4f98"}}}}(null));