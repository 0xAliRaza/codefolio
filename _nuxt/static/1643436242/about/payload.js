__NUXT_JSONP__("/about", (function(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s){return {data:[{about:{slug:"about",toc:[],body:{type:"root",children:[{type:b,tag:l,props:{},children:[{type:a,value:"And this is some text.\nConsequat aliquip incididunt adipisicing consequat enim velit magna sint magna esse. Quis mollit commodo velit ipsum velit ut veniam aliqua ipsum cupidatat culpa consectetur. Lorem eu et aute officia esse anim exercitation sit consequat cupidatat sunt Lorem aliquip. "},{type:b,tag:"strong",props:{},children:[{type:a,value:"Tempor eu"}]},{type:a,value:" nulla reprehenderit ad consequat nulla in cupidatat incididunt ad excepteur. Enim ullamco cillum mollit pariatur adipisicing esse id tempor mollit excepteur sint esse. Excepteur nisi laboris nulla amet. Sit excepteur occaecat quis laborum elit anim."}]},{type:a,value:g},{type:b,tag:l,props:{},children:[{type:b,tag:"em",props:{},children:[{type:a,value:"Esse aliquip cillum reprehenderit"}]},{type:a,value:" ea cillum excepteur laborum voluptate nisi cupidatat duis. Velit commodo quis amet non. Ex reprehenderit sit commodo reprehenderit et do."}]},{type:a,value:g},{type:b,tag:"div",props:{className:["nuxt-content-highlight"]},children:[{type:b,tag:"pre",props:{className:["language-ts","line-numbers"]},children:[{type:b,tag:"code",props:{},children:[{type:b,tag:c,props:{className:[d,f]},children:[{type:a,value:"import"}]},{type:a,value:" something "},{type:b,tag:c,props:{className:[d,f]},children:[{type:a,value:"from"}]},{type:a,value:h},{type:b,tag:c,props:{className:[d,"string"]},children:[{type:a,value:"'something'"}]},{type:a,value:g},{type:b,tag:c,props:{className:[d,f]},children:[{type:a,value:"export"}]},{type:a,value:h},{type:b,tag:c,props:{className:[d,f]},children:[{type:a,value:"class"}]},{type:a,value:h},{type:b,tag:c,props:{className:[d,"class-name"]},children:[{type:a,value:"Test"}]},{type:a,value:h},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:m}]},{type:a,value:j},{type:b,tag:c,props:{className:[d,"comment"]},children:[{type:a,value:"\u002F* Officia consectetur ipsum voluptate consectetur cillum. Velit culpa fugiat laborum sit ipsum reprehenderit exercitation ullamco reprehenderit amet amet aliquip cupidatat. Ex fugiat amet aute qui ipsum ullamco culpa. Exercitation ad ex est officia laborum Lorem anim consectetur do.\nVelit nulla aute cupidatat nulla cupidatat quis. Nostrud excepteur dolore exercitation fugiat officia culpa quis tempor eu. Cillum velit amet ut excepteur tempor nisi incididunt do aliquip aliquip amet occaecat dolore esse. Reprehenderit pariatur esse labore sunt eu esse dolore voluptate voluptate reprehenderit nisi. Consequat nisi ut non id eiusmod Lorem ex. *\u002F"}]},{type:a,value:j},{type:b,tag:c,props:{className:[d,f]},children:[{type:a,value:"public"}]},{type:a,value:" test"},{type:b,tag:c,props:{className:[d,"operator"]},children:[{type:a,value:":"}]},{type:a,value:" String\n  "},{type:b,tag:c,props:{className:[d,n]},children:[{type:a,value:"constructor"}]},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:o}]},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:p}]},{type:a,value:h},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:m}]},{type:a,value:"\n    "},{type:b,tag:c,props:{className:[d,"builtin"]},children:[{type:a,value:"console"}]},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:q}]},{type:b,tag:c,props:{className:[d,n]},children:[{type:a,value:"log"}]},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:o}]},{type:b,tag:c,props:{className:[d,f]},children:[{type:a,value:"this"}]},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:q}]},{type:a,value:"test"},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:p}]},{type:a,value:j},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:r}]},{type:a,value:g},{type:b,tag:c,props:{className:[d,e]},children:[{type:a,value:r}]},{type:a,value:g}]}]}]}]},dir:s,path:"\u002Fabout",extension:".md",createdAt:i,updatedAt:i},navbar:{slug:"navbar",resumeUrl:k,externalLinks:[{text:"github",url:k},{text:"blog",url:k}],dir:s,path:"\u002Fnavbar",extension:".json",createdAt:i,updatedAt:i}}],fetch:{},mutations:void 0}}("text","element","span","token","punctuation","keyword","\n"," ","2022-01-29T06:03:08.056Z","\n  ","#","p","{","function","(",")",".","}","\u002F")));