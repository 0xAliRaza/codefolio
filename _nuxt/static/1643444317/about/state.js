window.__NUXT__=(function(a){return {staticAssetsBase:"\u002Fcodefolio\u002F_nuxt\u002Fstatic\u002F1643444317",layout:"default",error:a,serverRendered:true,routePath:"\u002Fabout",config:{_app:{basePath:"\u002Fcodefolio\u002F",assetsPath:"\u002Fcodefolio\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"eafe4fa5"}}}}(null));