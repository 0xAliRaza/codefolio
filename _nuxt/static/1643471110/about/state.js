window.__NUXT__=(function(a){return {staticAssetsBase:"\u002Fcodefolio\u002F_nuxt\u002Fstatic\u002F1643471110",layout:"default",error:a,serverRendered:true,routePath:"\u002Fabout",config:{_app:{basePath:"\u002Fcodefolio\u002F",assetsPath:"\u002Fcodefolio\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"42d3b184"}}}}(null));