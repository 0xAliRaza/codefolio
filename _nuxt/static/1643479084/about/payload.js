__NUXT_JSONP__("/about", (function(a,b,c,d,e,f,g,h,i){return {data:[{about:{slug:"about",toc:[],body:{type:"root",children:[{type:d,tag:e,props:{},children:[{type:a,value:"Consequat aliquip 👋 incididunt adipisicing consequat enim velit magna sint magna esse. Quis mollit commodo velit ipsum velit ut veniam aliqua ipsum cupidatat culpa consectetur. Lorem 😃😃 eu et aute officia esse anim exercitation sit consequat cupidatat sunt Lorem aliquip. "},{type:d,tag:"strong",props:{},children:[{type:a,value:"Tempor eu"}]},{type:a,value:" nulla reprehenderit ad consequat nulla in cupidatat incididunt ad excepteur. Enim ullamco cillum mollit pariatur adipisicing esse id tempor mollit excepteur sint esse. Excepteur nisi laboris nulla amet. Sit excepteur occaecat quis laborum elit anim."}]},{type:a,value:g},{type:d,tag:e,props:{},children:[{type:d,tag:"em",props:{},children:[{type:a,value:"Esse aliquip cillum reprehenderit"}]},{type:a,value:" ea cillum 👽 excepteur laborum voluptate nisi cupidatat duis. Velit commodo quis amet non. Ex reprehenderit sit commodo reprehenderit et do."}]},{type:a,value:g},{type:d,tag:e,props:{},children:[{type:a,value:"Laborum nisi Lorem mollit laboris culpa non sunt sint excepteur. Cillum dolore Lorem consequat velit ullamco nisi ad irure exercitation qui amet. Amet tempor quis pariatur consequat est dolor consequat non id aliquip. Labore elit excepteur sint nostrud. Deserunt aute irure esse aliquip cupidatat ullamco non laborum labore. Dolor exercitation 🐱‍👤 Lorem ipsum qui pariatur culpa deserunt adipisicing non pariatur. Labore et est id adipisicing pariatur ea adipisicing aliquip ullamco incididunt ullamco consectetur 😉."}]}]},dir:f,path:"\u002Fabout",extension:".md",createdAt:c,updatedAt:c},navbar:{slug:"navbar",resumeUrl:"https:\u002F\u002Fexample.com\u002Fresume",externalLinks:[{text:h,url:b},{text:"blog",url:"https:\u002F\u002Fexample.com\u002Fblog"}],dir:f,path:"\u002Fnavbar",extension:i,createdAt:c,updatedAt:c},socialIcons:{slug:"social-icons",icons:[{name:"discord",url:b},{name:"facebook",url:b},{name:h,url:b},{name:"twitter",url:b},{name:"gmail",url:b},{name:"skype",url:b}],dir:f,path:"\u002Fsocial-icons",extension:i,createdAt:c,updatedAt:c}}],fetch:{},mutations:void 0}}("text","#","2022-01-29T17:57:16.033Z","element","p","\u002F","\n","github",".json")));