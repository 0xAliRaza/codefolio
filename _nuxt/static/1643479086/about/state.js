window.__NUXT__=(function(a){return {staticAssetsBase:"\u002Fcodefolio\u002F_nuxt\u002Fstatic\u002F1643479086",layout:"default",error:a,serverRendered:true,routePath:"\u002Fabout",config:{_app:{basePath:"\u002Fcodefolio\u002F",assetsPath:"\u002Fcodefolio\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"3dcd593b"}}}}(null));