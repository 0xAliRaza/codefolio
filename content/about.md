Consequat aliquip 👋 incididunt adipisicing consequat enim velit magna sint magna esse. Quis mollit commodo velit ipsum velit ut veniam aliqua ipsum cupidatat culpa consectetur. Lorem 😃😃 eu et aute officia esse anim exercitation sit consequat cupidatat sunt Lorem aliquip. **Tempor eu** nulla reprehenderit ad consequat nulla in cupidatat incididunt ad excepteur. Enim ullamco cillum mollit pariatur adipisicing esse id tempor mollit excepteur sint esse. Excepteur nisi laboris nulla amet. Sit excepteur occaecat quis laborum elit anim.

_Esse aliquip cillum reprehenderit_ ea cillum 👽 excepteur laborum voluptate nisi cupidatat duis. Velit commodo quis amet non. Ex reprehenderit sit commodo reprehenderit et do.

Laborum nisi Lorem mollit laboris culpa non sunt sint excepteur. Cillum dolore Lorem consequat velit ullamco nisi ad irure exercitation qui amet. Amet tempor quis pariatur consequat est dolor consequat non id aliquip. Labore elit excepteur sint nostrud. Deserunt aute irure esse aliquip cupidatat ullamco non laborum labore. Dolor exercitation 🐱‍👤 Lorem ipsum qui pariatur culpa deserunt adipisicing non pariatur. Labore et est id adipisicing pariatur ea adipisicing aliquip ullamco incididunt ullamco consectetur 😉.
