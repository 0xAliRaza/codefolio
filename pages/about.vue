<template>
  <div>
    <Header
      :resume-url="navbar.resumeUrl"
      :external-links="navbar.externalLinks"
    />
    <main class="main">
      <div class="container">
        <div class="row justify-content-center">
          <div class="col-12 col-lg-10 col-xl-8">
            <section class="about">
              <div class="about__heading my-3 text-center">
                <h1>About</h1>
              </div>
              <div class="p-3">
                <nuxt-content :document="about" />
              </div>
              <div class="about__links">
                <a
                  v-for="(icon, i) in socialIcons.icons"
                  :key="icon.name + i"
                  class="about__link"
                  :href="icon.url"
                  target="_blank"
                >
                  <Icon class="about__icon" :name="icon.name" />
                </a>
              </div>
              <div
                class="
                  my-3
                  text-nowrap
                  d-flex
                  align-items-center
                  justify-content-center
                "
              >
                <Button
                  v-if="navbar.resumeUrl"
                  primary-glow
                  :link="navbar.resumeUrl"
                  class="me-1"
                  download=""
                  >Resume <Icon class="ms-1" name="download"
                /></Button>
                <Button outline-primary nuxt-link="/contact">Contact</Button>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
    <Footer />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  async asyncData({ $content }) {
    const about = await $content('about').fetch()
    const navbar = await $content('navbar').fetch()
    const socialIcons = await $content('social-icons').fetch()
    return {
      about,
      navbar,
      socialIcons,
    }
  },
  head: {
    title: 'About - Codefolio',
  },
})
</script>
<style lang="scss" scoped>
@import '@/scss/abstracts';
.about {
  padding: pxToRem(10) 0;

  &__links {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    grid-gap: pxToRem(8);
    padding: pxToRem(20);
    z-index: 200;
  }
  &__link {
    backface-visibility: hidden;
    transform: translateZ(0);
    display: inline-flex;
    color: $gray-7;
    transition: all 0.2s ease;
    @include hoctive {
      transform: scale(1.2);
      color: $gray-8;
    }
  }
  &__icon {
    width: 2em !important;
    height: 2em !important;
  }
}
</style>
