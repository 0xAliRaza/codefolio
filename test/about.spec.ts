import { shallowMount } from '@vue/test-utils'
import about from '@/pages/about.vue'

describe('about', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(about, {
      data() {
        return {
          navbar: { resumeUrl: '', externalLinks: [] },
          about: {},
          socialIcons: {
            icons: [
              { name: 'discord', url: '#' },
              { name: 'twitter', url: '#' },
              { name: 'facebook', url: '#' },
            ],
          },
        }
      },
      stubs: ['Icon', 'nuxt-content'],
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
