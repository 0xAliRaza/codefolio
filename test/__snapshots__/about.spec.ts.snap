// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about renders correctly 1`] = `
<div>
  <header resume-url
          external-links
  >
  </header>
  <main class="main">
    <div class="container">
      <div class="row justify-content-center">
        <div class="col-12 col-lg-10 col-xl-8">
          <section class="about">
            <div class="about__heading my-3 text-center">
              <h1>
                About
              </h1>
            </div>
            <div class="p-3">
              <nuxt-content-stub document="[object Object]">
              </nuxt-content-stub>
            </div>
            <div class="about__links">
              <a href="#"
                 target="_blank"
                 class="about__link"
              >
                <icon-stub name="discord"
                           class="about__icon"
                >
                </icon-stub>
              </a>
              <a href="#"
                 target="_blank"
                 class="about__link"
              >
                <icon-stub name="twitter"
                           class="about__icon"
                >
                </icon-stub>
              </a>
              <a href="#"
                 target="_blank"
                 class="about__link"
              >
                <icon-stub name="facebook"
                           class="about__icon"
                >
                </icon-stub>
              </a>
            </div>
            <div class="
                my-3
                text-nowrap
                d-flex
                align-items-center
                justify-content-center
              ">
              <button outline-primary
                      nuxt-link="/contact"
              >
                Contact
              </button>
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
  <footer>
  </footer>
</div>
`;
